ex5/ex5.c

```diff
@@ -15,7 +15,29 @@ char* msg3 = "hello world #3";
 
 int main()
 {
-    // Your code here
-    
+    char inbuf[MSGSIZE];
+    int p[2];
+
+    if (pipe(p) < 0){
+        fprintf(stderr, "pipe failed\n");
+        exit(1);
+    };
+
+    int child = fork();
+
+    if (child < 0){
+        fprintf(stderr, "fork failed\n");
+        exit(1);
+    } else if (child == 0){
+        write(p[1], msg1, MSGSIZE);
+        write(p[1], msg2, MSGSIZE);
+        write(p[1], msg3, MSGSIZE);
+    } else {
+        for (int i = 0; i < 3; i++){
+            read(p[0], inbuf, MSGSIZE);
+            printf("%s \n", inbuf);
+        }
+    }
+
     return 0;
 }
```

Comment:
Your exercises look great, keep up the good work! I notice you do not have exercise 6 done yet. When you do get started, make sure you utilize write(). I noticed some students having trouble with this problem; this line of code might help you get started:
write(STDOUT_FILENO, NULL, sizeof(NULL));